receiver/receiver.py

```diff
@@ -52,13 +57,16 @@ def read_video_parameters() -> (int, int, int):
         raise AssertionError('bug')
 
     untrusted_input = os.read(0, input_size)
+    if not untrusted_input:
+        raise RuntimeError('can not read from stream')
     if len(untrusted_input) != input_size:
         raise RuntimeError('wrong number of bytes read')
     untrusted_width, untrusted_height, untrusted_fps = s.unpack(untrusted_input)
     del untrusted_input
 
-    if untrusted_width > 4096 or untrusted_height > 4096 or untrusted_fps > 4096:
-        raise RuntimeError('excessive width, height, and/or fps')
+    screen = Gdk.Display().get_default().get_default_screen()
+    if untrusted_width > screen.width() or untrusted_height > screen.height() or untrusted_fps > 4096:
+        print('warning: excessive width, height, and/or fps')
```

Comment:
`untrusted_fps` being too large is definitely an error.  I _really_ do not want to allow `untrusted_width * untrusted_height` that exceeds `INT_MAX / 4`, or other values that risk causing integer overflow downstream.